app/scripts/models/auth_brokers/base.js

```diff
@@ -175,6 +186,15 @@ define([
 
     _isForceAuthUrl: function () {
       return this.window.location.pathname === '/force_auth';
+    },
+
+    _isSignInUrl: function () {
+      var pathname = this.window.location.pathname;
+      return pathname === '/signin' || pathname === '/oauth/signin';
```

Comment:
Following the line of thinking [here](https://github.com/mozilla/fxa-content-server/pull/2019/files#r22624336), we could just add another action, say `action=force_signin` or `action=force_auth`, which would require an `email` param, and the OAuth server could redirect to `/force_auth` with an email param. I think that would make the semantics cleaner and remove the need for many of these awkward if statements here.